src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.ILLink.targets

```diff
@@ -109,12 +119,14 @@ Copyright (c) .NET Foundation. All rights reserved.
 
   <!--
     ============================================================
-                     _SetILLinkDefaults
+                     PrepareForILLink
 
-    Set up the default options and inputs to ILLink.
+    Set up the default options and inputs to ILLink. Other targets are expected to hook into
+    this extension point via BeforeTargets/AfterTargets to opt assemblies into or out of trimming,
```

Comment:
Another target can't add/remove assemblies from `@(ManagedAssemblyToLink)`, correct? They can only change the metadata (IsTrimmable, TrimMode) on the existing assemblies.

Reply:
Correct. Technically they do so, but it's unsupported - it wouldn't work with the rest of the publish pipeline. I'll make a note here.

Reply:
I suppose we could change the implementation to support this in case people wanted to add linker inputs that aren't normally a part of publish, or exclude some assemblies from being analyzed by the linker but still have them be processed by `PublishReadyToRun`.

Normally though, I'd recommend changing the publish set instead. I generally prefer to restrict what we officially support until there is a more obvious scenario we can use to check the behavior.

Reply:
That makes sense to me. Let’s wait to see if it’s needed.